api/lib/application/campaigns/campaign-controller.js

```diff
@@ -67,9 +67,25 @@ module.exports = {
   update(request) {
     const { userId } = request.auth.credentials;
     const campaignId = parseInt(request.params.id);
-    const { title, 'custom-landing-page-text': customLandingPageText } = request.payload.data.attributes;
+    const { title, 'custom-landing-page-text': customLandingPageText, 'archived-at': archivedAt } = request.payload.data.attributes;
 
-    return usecases.updateCampaign({ userId, campaignId, title, customLandingPageText })
+    return usecases.updateCampaign({ userId, campaignId, title, customLandingPageText, archivedAt })
+      .then(campaignSerializer.serialize);
+  },
+
+  archiveCampaign(request) {
+    const { userId } = request.auth.credentials;
+    const campaignId = parseInt(request.params.id);
+
+    return usecases.updateCampaign({ userId, campaignId, archivedAt: new Date() })
```

Comment:
ce ne serait pas plus pertinent d'avoir un usecase spécifique pour notre cas ?

Reply:
On s'est posé la question mais on trouvait que ça faisait un peu overkill vu la simplicité. Si demain on se retrouve à avoir un archivage / désarchivage qui a plus de métier, à ce moment là on pourra refactor et en faire `usecase` séparé

Reply:
Ben il s'agit quand même d'un cas d'utilisation bien distinct ! En réutilisant le précédent on dirait que tu traites le _usecase_ comme une boîte à outils alors qu'il s'agit de refléter le métier…

Reply:
Aller je vous fais une confession... au début quand on était partie sur un `PATCH` du front et qu'on s'était rendu compte de l'erreur, il a fallu défaire beaucoup de choses, et j'avais une flemme intersidérale de créer deux usecase passe-plat pour ça, même si je sentais que c'était la bonne chose à faire... :D YOU WIN